### PR TITLE
Different csv headers by bench mode on infra error

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #103135
* __->__ #103134

As title. The headers are different for distinct bench mode. This PR is a supplement
to https://github.com/pytorch/pytorch/pull/100372 to respect `performance` mode where numerical speedup is expected
instead of status text.

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @aakhundov

### CODE REVIEW

**Review by thiagocrepaldi — COMMENTED** [review]
Please consider adding description to the PR. The ghstack does not tell why the PR is needed and how the problem is solved

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -2743,12 +2743,27 @@ def detect_and_mark_batch(t):
...
+                    rows = [
+                        [device, name, placeholder_batch_size, 0.0, 0.0]
+                        for device in args.devices
+                    ]
+                else:
+                    headers = []
+                    rows = [
+                        [device, name, placeholder_batch_size, 0.0]
+                        for device in args.devices
+                    ]
```
[the remark this answers is not in the document]
Comment (BowenBao) [reply benchmarks/dynamo/common.py:2763]: Headers were added by previous commit. Before that it doesn't. I'm happy to drop this branch if things are already adapted to the new format. However the branch on "performance" is important as without it we get wrong header when first model for perf test fails.

**Review by thiagocrepaldi — APPROVED** [review]
LGTM

File: benchmarks/dynamo/common.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (thiagocrepaldi) [reply benchmarks/dynamo/common.py:2763]: no preference for me. I was just checking whether an unexpected behavior was being introduced

**Comment by BowenBao** [comment]
@ezyang please kindly take a look.

**Review by ezyang — APPROVED** [review]
oh, nice catch, thanks. I guess we should probably refactor this code eventually..

**Pull request closed without merging**